app/uk/gov/hmrc/merchandiseinbaggagefrontend/controllers/DeclarationConfirmationController.scala

```diff
@@ -27,18 +30,22 @@ import scala.concurrent.{ExecutionContext, Future}
 class DeclarationConfirmationController @Inject()(
                                                    override val controllerComponents: MessagesControllerComponents,
                                                    actionProvider: DeclarationJourneyActionProvider,
-                                                   view: DeclarationConfirmationView
+                                                   view: DeclarationConfirmationView,
+                                                   repo: DeclarationJourneyRepository
                                                  )(implicit ec: ExecutionContext, appConf: AppConfig)
-  extends DeclarationJourneyUpdateController with MibReferenceGenerator {
+  extends DeclarationJourneyController with MibReferenceGenerator {
 
   val onPageLoad: Action[AnyContent] = actionProvider.journeyAction.async { implicit request =>
-    val reference = mibReference.toOption
     request.declarationJourney.declarationIfRequiredAndComplete.fold(actionProvider.invalidRequestF) { declaration =>
-      Future.successful(Ok(view(declaration.copy(mibReference = reference))))
+      resetJourney.map(ref => Ok(view(declaration.copy(mibReference = Some(ref)))))
     }
   }
 
-  val onSubmit: Action[AnyContent] = actionProvider.journeyAction.async { implicit request =>
-    Future.successful(Ok("onSubmit"))
+  private def resetJourney(implicit request: DeclarationJourneyRequest[AnyContent]): Future[MibReference] = {
+    import request.declarationJourney._
+    for {
+      _         <- repo.upsert(DeclarationJourney(sessionId, declarationType))
+      reference <- Future.fromTry(mibReference)
```

Comment:
Why is the Try needed? How can MibReference creation fail?

Also MibReference is still optional on the Declaration - it should be mandatory.